app/controllers/subscriptions_controller.rb

```diff
@@ -30,6 +29,14 @@ def show
 
   private
 
+  def create_subscription(subscription_params, iugu_id)
+    Subscription.create!(subscription_params.merge(
+      user: current_user,
+      iugu_id: iugu_id,
+      affiliate_tag: affiliate_tag
```

Comment:
I wouldn't create an `affiliate_tag` method. Instead, I'd just use `affiliate_tag: request.env['affiliate.tag'].presence` here :)

(more on that below)